### PR TITLE
Remove the LLL compiler

### LINKED ISSUE #7949: Remove LLL from Solidity
Reason explained [here](https://github.com/ethereum/solidity/issues/1230#issuecomment-559116361)

TODO:
- [ ] Remove liblll and lllc
- [ ] Remove from documentation
- [ ] Remove tests
- [ ] Remove cmake settings
- [ ] Remove form CI
- [ ] Add link to "assembler" in documentation

Closes #1188
Closes #1230
Closes #1288
Closes #1453
Closes #1478
Closes #2865
Closes #2932
Closes #3141
Closes #4680
Closes #5714

### PR DESCRIPTION
**Pull request by axic**
Closes #7949.

### CODE REVIEW

**Comment by axic** [comment]
There's more to remove from scripts, etc.

**Comment by chriseth** [comment]
Requires rebase.

**Comment by leonardoalt** [comment]
> There's more to remove from scripts, etc.

Is that supposed to be done in this PR or later?

**Comment by axic** [comment]
> > There's more to remove from scripts, etc.
>
> Is that supposed to be done in this PR or later?

Here, just didn't have time yesterday to finish.

**Comment by axic** [comment]
@leonardoalt @chriseth updated, should be ready now

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,11 +19,6 @@ if (IS_BIG_ENDIAN)
 	message(FATAL_ERROR "${PROJECT_NAME} currently does not support big endian systems.")
 endif()
 
-option(LLL "Build LLL" OFF)
-option(SOLC_LINK_STATIC "Link solc executable statically on supported platforms" OFF)
```
Comment (leonardoalt) [inline CMakeLists.txt:23]: Why is this removed?

File: libevmasm/Assembly.cpp [hunk 2]
```diff
@@ -38,48 +38,6 @@ using namespace solidity::evmasm;
 using namespace solidity::langutil;
 using namespace solidity::util;
 
-void Assembly::append(Assembly const& _a)
```
Comment (leonardoalt) [inline libevmasm/Assembly.cpp:41]: Why are these removed?

Comment (leonardoalt) [reply libevmasm/Assembly.cpp:41]: Ah, found the answer :p

**Review by leonardoalt — COMMENTED** [review]
Just one question left

**Comment by leonardoalt** [comment]
One test failing

File: CMakeLists.txt (hunk 1 above)
Comment (axic) [reply CMakeLists.txt:23]: To validate that you guys are still reviewing properly.

Comment (leonardoalt) [reply CMakeLists.txt:23]: My morning reviews are proper, afternoon questionable

**Comment by axic** [comment]
Fixed the test expectations.

File: CMakeLists.txt (hunk 1 above)
Comment (axic) [reply CMakeLists.txt:23]: It is already the afternoon. Should I not trust the review? :)

**Review by leonardoalt — APPROVED** [review]

File: libevmasm/Assembly.cpp (hunk 2 above)
Comment (chriseth) [inline libevmasm/Assembly.cpp:41]: Nice!

**Review by chriseth — APPROVED** [review]
